### PR TITLE
Feat/windows nodes release

### PR DESCRIPTION
**Pull request by wujunyi792**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Added support for windows/amd64 release, including components edgecore and keadm

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
This pr depends on #4968
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: keadm/cmd/keadm/app/cmd/common/constant_windows.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
+//go:build windows
+
+/*
+Copyright 2019 The KubeEdge Authors.
```
Comment (Shelley-BaoYue) [inline keadm/cmd/keadm/app/cmd/common/constant_windows.go:4]:
```suggestion
Copyright 2023 The KubeEdge Authors.
```

File: keadm/cmd/keadm/app/cmd/util/common_windows.go [hunk 2]
```diff
@@ -0,0 +1,249 @@
+//go:build windows
+
+/*
+Copyright 2019 The KubeEdge Authors.
```
Comment (Shelley-BaoYue) [inline keadm/cmd/keadm/app/cmd/util/common_windows.go:4]:
```suggestion
Copyright 2023 The KubeEdge Authors.
```

**Comment by ZhengXinwei-F** [comment]
This PR appears to combine linux and windows compiling into a single function.

This will make future support for more OS or Windows arm compilation more challenging.Too many if branches will also degrade code readability. 

Perhaps you can separate the logic for different operating systems into independent functions.
Like:
```
function build_kubeedge_release() {
    case $OS in
    linux）
    build_kubeedge_release_linux
    ;;
    windows）
    build_kubeedge_release_windows
    ;; 
    *）
    echo "not support OS $OS"
    exit 1
    ;; 
    esac 

    exit 0
}
```

**Comment by wujunyi792** [comment]
> This PR seems to implement linux and windows compilation in a single function.
> 
> This will make future support for more OS or windows-arm more difficult. Too many if branches will also degrade code readability.

Yes, it's necessary to address the problem. However, since it might be a significant task, I've opened an issue to investigate further. #5124

**Review by fisherxu — COMMENTED** [review]
/approve
/assign @Shelley-BaoYue

**Pull request merged**